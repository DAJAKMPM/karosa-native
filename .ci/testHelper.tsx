import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
