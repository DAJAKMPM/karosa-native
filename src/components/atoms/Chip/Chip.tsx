/**
 *
 * Chip
 * @format
 *
 */

import React, { FC } from "react";
import { Chip as RnChip } from "react-native-elements";
import Icon from "@app/atoms/Icon";

import type { PropsType } from "./types";
import { ICON_SIZE } from "./config";
import ChipStyles from "./styles";

const Chip: FC<PropsType> = (props) => {
  const { selected, onPress, title, iconName, iconGroup } = props;

  return (
    <RnChip
      onPress={onPress}
      title={title}
      buttonStyle={
        selected ? ChipStyles.selectedContainer : ChipStyles.unselectedContainer
      }
      titleStyle={selected ? ChipStyles.txtSelected : ChipStyles.txtUnselected}
      type="outline"
      icon={
        <Icon
          group={iconGroup}
          name={iconName}
          height={ICON_SIZE}
          width={ICON_SIZE}
          extraStyle={ChipStyles.iconContainer}
        />
      }
    />
  );
};

export default Chip;
