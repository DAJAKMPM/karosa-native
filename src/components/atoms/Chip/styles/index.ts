/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";
import { theme } from "@app/styles";

const ChipStyles = StyleSheet.create({
  selectedContainer: {
    borderRadius: 3,
    borderColor: theme.colors.primary,
    backgroundColor: theme.colors.green15,
    marginRight: 8,
    marginTop: 8,
  },
  unselectedContainer: {
    borderRadius: 3,
    borderColor: theme.colors.dark30,
    marginRight: 8,
    marginTop: 8,
  },
  txtSelected: {
    fontWeight: "700",
    color: theme.colors.primary,
  },
  txtUnselected: {
    color: theme.colors.dark30,
  },
  iconContainer: {
    marginRight: 8,
  },
});

export default ChipStyles;
