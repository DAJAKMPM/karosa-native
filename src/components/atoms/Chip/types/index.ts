/**
 *
 * @format
 *
 */

import { ReactElement } from "react";

export type PropsType = {
  title: string | ReactElement;
  selected: boolean;
  onPress?(): void;
  iconName?: string;
  iconGroup?: string;
  width?: number;
};
