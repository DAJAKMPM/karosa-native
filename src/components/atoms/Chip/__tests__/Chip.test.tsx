import React from "react";
import { renderWithWrapper } from "@app/ci/testHelper";
import Chip from "../Chip";

describe("Chip Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(<Chip selected title="test" />);

    expect(toJSON()).toMatchSnapshot();
  });
});
