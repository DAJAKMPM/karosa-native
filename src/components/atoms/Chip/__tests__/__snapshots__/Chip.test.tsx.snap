// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#61D49726",
          "borderColor": "#0AA351",
          "borderRadius": 3,
          "borderWidth": 0.5,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginRight": 8,
          "marginTop": 8,
          "padding": 8,
          "width": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#0AA351",
            "fontSize": 14,
            "fontWeight": "700",
            "paddingHorizontal": 2,
            "paddingVertical": 1,
            "textAlign": "center",
          }
        }
      >
        test
      </Text>
    </View>
  </View>
</View>
`;
